modules/storage/src/storage-provider/providers/aliyun/index.ts

```diff
@@ -0,0 +1,204 @@
+import { IStorageProvider, StorageConfig } from '../../interfaces';
+
+import OSS from 'ali-oss';
+
+import fs from 'fs';
+
+import { basename } from 'path';
+
+export class AliyunStorage implements IStorageProvider {
+  private _activeContainer: string = '';
+  private _ossClient: OSS;
+
+  constructor(options: StorageConfig) {
+    const conf = options.aliyun;
+    if (!conf || !conf.accessKeyId || !conf.accessKeySecret || !conf.region) {
+      throw Error('aliyun config is invalid');
+    }
+
+    this._ossClient = new OSS({
+      region: conf.region,
+      accessKeyId: conf.accessKeyId,
+      accessKeySecret: conf.accessKeySecret,
+    });
+  }
+
+  async containerExists(name: string): Promise<boolean | Error> {
+    return await this._ossClient
+      .getBucketInfo(name)
+      .then(() => true)
+      .catch(err => {
+        if (err.status === 404) {
+          return false;
+        }
+
+        throw err;
+      });
+  }
+
+  async createContainer(name: string): Promise<boolean | Error> {
+    await this._ossClient.putBucket(name);
+    this.container(name);
+    return true;
+  }
+
+  container(name: string): IStorageProvider {
+    this._activeContainer = name;
+    this._ossClient.useBucket(name);
+    return this;
+  }
+
+  async deleteContainer(name: string): Promise<boolean | Error> {
+    await this._ossClient.deleteBucket(name);
+    return true;
+  }
+
+  async createFolder(name: string): Promise<boolean | Error> {
+    await this._ossClient.put(name, Buffer.from(''));
+    return true;
+  }
+
+  async folderExists(name: string): Promise<boolean | Error> {
+    return await this._ossClient
```

Comment:
this is the same as file exists. So it's better to call the "exists" function and not duplicate the code.